### DISCUSSION TITLE
Unstable glowstone canyon sand should give the player a small warning before falling

### DISCUSSION THREAD

**Discussion by Ramoddi** [infernalstudios/Infernal-Expansion#166] [Ideas]
**Describe the addition:**
Warning for the unstable sand

**Additional Context:**
One of the rules of Minecraft is that the player should be at fault for dying. I think falling to their deaths, often in lava, just by walking should not be a thing. I would love this feature if the sand glimmers and makes a sound or something, to warn the player of the danger. This would make the feature more fun and fair! And make it easier to clutch, which could be satisfying yet andrealin filled.

**Comment by Hellion-Games** [comment]
I know what you mean, before we continue with this discussion, I do want to point out that the falling blocks have a different (and usually darker) texture than the other sand, which is a fairly good visual cue to avoid it.

That said, do you still think that isn't enough of a warning?

**Comment by Ramoddi** [comment]
I plopped straight into lava with no idea what happened. Also when going around in creative I did not notice the difference, there is just som much new fascinating stuff to look at that I do not notice the different texture. I think this would take at least once, or multiple falls before a player learns the dangers. The problem I feel, is that I cannot save myself and I ended up confused instead of feeling like it was my fault. I love the idea of the feature though! And think with a small warning for the player, it could be one of the coolest features I have seen in modded biomes

**Reply by Hellion-Games** [reply]
I'm curious, what would you think if the falling blocks emitted a small amount of particles so that there is some sort of movement to draw the eye? It's a matter of not wanting to make it _too_ obvious while still being noticeable. I've noticed the texture can be hard to see, but if you know to look for it, it's pretty obvious. I think a small amount of particles might help reinforce that.

**Comment by Hellion-Games** [comment]
The other interesting thing is that I don't believe those should ever drop into lava. They should always have a solid bottom and not be a large enough fall to kill at full health. We will have to look into that and maybe think about some sort of sensory warning!

**Comment by Ramoddi** [comment]
Oo, then it might be a bug!
(Though, I still think a small warning would lead to a bit better gameplay)

Here is a picture of the hole I fell into
![bilde](https://user-images.githubusercontent.com/79367831/125858932-aaecbe31-8d9b-4a90-a4ed-248389e724fb.png)
'When I look at it again I notice it is weirdly square-ish?

**Reply by Hellion-Games** [reply]
They are meant to be somewhat square, but that one does seem to be a bit to the extreme. Have you found other similar ones that fall into lava?

**Reply by Ramoddi** [reply]
Exploring around, not similar square ones, but some on edges with lava
![bilde](https://user-images.githubusercontent.com/79367831/125865382-645d2de4-8366-459e-84da-eeaca391f527.png)
![bilde](https://user-images.githubusercontent.com/79367831/125865402-a9366d22-523e-40c7-b54b-a98d3a47e8c5.png)
![bilde](https://user-images.githubusercontent.com/79367831/125865552-7fe1d9a7-136a-47c8-bdd0-54b9d15cefbf.png)

I notice some of them falling when I have the Luminous effect too.

**Reply by Hellion-Games** [reply]
Wait. Are these the traps or just floating generation of glowsand?

**Reply by Hellion-Games** [reply]
These look more like just floating sand generating. The Luminous activating would probably be from a block update from the fungus.

**Reply by Ramoddi** [reply]
Maybe just normal glowsand. They fall when the player nears with Luminous effect, which can cause a jump to them to be critically dangerous. I think the effect makes them update. This probably should not happen, as if I have a Luminous effect, and I jump or run to a place with the sand, it falls

**Reply by Hellion-Games** [reply]
It shouldn't be the effect itself. Luminous fungus are motion detecting. They change blockstates (light up) when something moves near them. That means there is a block update, so they would drop floating sand if they're on it. It doesn't actually have to do with the effect. The fungus can also be triggered safely from a distance with projectiles. So, this actually doesn't have to do with the trapped glowdust sand after all which is kinda funny. I don't know if there is a good way to fix this issue in any way. Mostly just be careful around ledges where sand is floating.

**Reply by nekomaster1000** [reply]
Here's an idea: I don't know if it'd be irritating to implement, but what if Shimmer Sand jittered before it was about to fall? In the same way Piglins do when they're about to Zombify, but with a block. That gives you a fairer warning then!

**Reply by Hellion-Games** [reply]
It may be a pain to implement because they're blocks rather than entities, but could be interesting.

**Reply by Ramoddi** [reply]
Yeah, something like what nekomaster1000 is saying I think would be the best implementation! 😁

**Reply by Ramoddi** [reply]
Also a cute little jitter/shimmer sound 🥺

**Reply by Hellion-Games** [reply]
Yup! The plan is delayed fall, sound plays, and particles shoot out a bit.

**Reply by Ramoddi** [reply]
Superb! 😍
